### ISSUE TITLE
Using ubuntu-latest for GHAction Test

### ISSUE THREAD

**Issue by Blackjacx** [kareman/SwiftShell#92]
Hey there,

I asked myself why you cannot use ubuntu-latest for the [github action workflow](https://github.com/kareman/SwiftShell/blob/master/.github/workflows/test.yml) but instead `ubuntu-18.04`? I integrated your framework now and after this [my GH action](https://github.com/Blackjacx/Assist/actions/runs/271564850) is not green anymore 😢

**Comment by kareman** [comment]
Swift 5.3 requires SwiftShell 5.1.0-beta.1 . I will release it as non-beta this week.

**Comment by stherold** [comment]
Nice 🎉  Really cool tool by the way :-)

Just a question for a possible improvement:
Why is `run(...)` not throwable too like runAndPrint? What I would like to avoid is the extra check for the error and manually print the error like this:
```swift
        let out = run("mint", args)

        if let error = out.error {
            print(out.stderror)
            throw error
        }
```

The following would be much cleaner imho and the failure reason could be part of the error or an underlying error maybe (error catched nd printed at higher level):
```swift
        let out = try run("mint", args)
```

**Comment by kareman** [comment]
Thank you.

You're right, it is a bit cumbersome the way it is now. The problem with your suggestion is that it makes it more cumbersome to get the result of a failing command. But your use case is probably more common.

 I've been wanting to improve the API for a long time now, but I just haven't been able to come up with a good solution that handles all aspects well, like error handling, synchronicity, shell interpreter, stdin, joined or separate stdout/stderror, printing output, piping output to another command, and probably even more that I haven't thought of yet.

**Comment by kareman** [comment]
I just released 5.1.0.

I'll close this issue now as the original issue has been resolved, but feel free to open new issues if you have further suggestions on how to improve the API.
